django/db/models/query_utils.py

```diff
@@ -281,6 +282,20 @@ def _check_parent_chain(self, instance):
             return getattr(instance, link_field.attname)
         return None
 
+    def fetch_one(self, instance):
+        instance.refresh_from_db(fields=[self.field.attname])
+
+    def fetch_many(self, instances):
+        attname = self.field.attname
+        value_by_pk = {
+            pk: value
+            for pk, value in self.field.model._base_manager.db_manager()
+            .filter(pk__in={i.pk for i in instances})
```

Comment:
Any interest in observing database parameter limits, similar to how `in_bulk()` [does it](https://github.com/django/django/blob/0ddbe12ea99a2dc1b757dc2015ba8bb6bfd9d653/django/db/models/query.py#L1195), or factoring out some reusable private helper?

Reply:
Kind of a shame that `in_bulk` doesn't support `values` / `values_list` in the first place as that could have been

```python
value_by_pk = (
    self.field.model._base_manager
    .values_list(attname, flat=True).in_bulk({i.pk for i in instances})
)
```

in all cases it seems that the `db_manager()` call is unnecessary?

Reply:
> Kind of a shame that in_bulk doesn't support values / values_list

I'll open a new features ticket and link to the [sketched](https://github.com/django/django/pull/18497) implementation, given that we have this additional use in Django itself now.

Reply:
Also, not sure if the dict literal is faster but the `dict` function can be used as well

```python
        value_by_pk = dict(
            self.field.model._base_manager
            .filter(pk__in={i.pk for i in instances})
            .values_list("pk", attname)
        )